operator/initc/internal/wait.go

```diff
@@ -149,7 +191,10 @@ func (c *ParentPodCliqueDependencies) WaitForReady(ctx context.Context, log logr
 	}
 }
 
+// createClient creates and returns a Kubernetes clientset using the in-cluster configuration.
+// This function is designed to be called from within a pod running in a Kubernetes cluster.
 func createClient() (*kubernetes.Clientset, error) {
+	// Get the in-cluster REST configuration
```

Comment:
no need for this comment. When code is simple and self-documentation additional documentation on top is just noise.